core/libp2p/transport/tcp/tcp_connection.cpp

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "libp2p/transport/tcp/tcp_connection.hpp"
+
+namespace libp2p::transport {
+
+  TcpConnection::TcpConnection(basic::yield_t &yield,
+                               boost::asio::ip::tcp::socket &&socket)
+      : yield_(yield), socket_(std::move(socket)) {}
+
+  TcpConnection::TcpConnection(basic::yield_t &yield)
+      : yield_(yield), socket_(yield.get_executor().context()) {}
```

Comment:
Can you use socket, which is not "related" to the passed `yield`?

Reply:
i don't understand

Reply:
I mean, in the previous constructor you pass yield and socket disctincly, while in this one you deduce one from another. Is it allowed to do it, like in the previous constructor? In that case socket can be from the other yield or something like that

Reply:
one constructor handy to create client, while the other one to create server. 

any combination is valid actually